// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.syntax = "proto2";

package quickstep.serialization;

import "expressions/window_aggregation/WindowAggregateFunction.proto";
import "expressions/Expressions.proto";

message WindowAggregationOperationState {
  required int32 input_relation_id = 1;
  required WindowAggregateFunction function = 2;
  repeated Scalar arguments = 3;
  repeated Scalar partition_by_attributes = 4;
  required bool is_row = 5;
  required int64 num_preceding = 6;  // -1 means UNBOUNDED PRECEDING.
  required int64 num_following = 7;  // -1 means UNBOUNDED FOLLOWING.
  repeated Scalar order_by_attributes = 8;
}
